### PR TITLE
fix(admin): notification values not saved

### PR DESCRIPTION
**Pull request by mattjoke**
* added binding to reflect change when user inputs data in notification

### CODE REVIEW

**Review by sarkapalkovicova — APPROVED** [review]
Works.

**Review by bodnara — CHANGES_REQUESTED** [review]
This does not seem to fix the issue. I think the problem is that the value is not updated/validated before creation. It is only done when saved, which is probably why editing works fine.

**Comment by xflord** [comment]
> This does not seem to fix the issue. I think the problem is that the value is not updated/validated before creation. It is only done when saved, which is probably why editing works fine.

yep, `create()` method needs to be updated similarly to `save()`

**Comment by mattjoke** [comment]
> > This does not seem to fix the issue. I think the problem is that the value is not updated/validated before creation. It is only done when saved, which is probably why editing works fine.
> 
> yep, `create()` method needs to be updated similarly to `save()`

HTML escaping is done on the backend side. The code you are referring to, is just a unused part of the code.

**Comment by xflord** [comment]
> > > This does not seem to fix the issue. I think the problem is that the value is not updated/validated before creation. It is only done when saved, which is probably why editing works fine.
> > 
> > 
> > yep, `create()` method needs to be updated similarly to `save()`
> 
> HTML escaping is done on the backend side. The code you are referring to, is just a unused part of the code used before the merge. And therefore the solution in this PR does not correlate to the problem.

I see, is it safe to remove the code then? Also, since the backend validation is back up again, do we plan to uncomment/fix frontend validation?

**Comment by mattjoke** [comment]
> > > > This does not seem to fix the issue. I think the problem is that the value is not updated/validated before creation. It is only done when saved, which is probably why editing works fine.
> > > 
> > > 
> > > yep, `create()` method needs to be updated similarly to `save()`
> > 
> > 
> > HTML escaping is done on the backend side. The code you are referring to, is just a unused part of the code used before the merge. And therefore the solution in this PR does not correlate to the problem.
> 
> I see, is it safe to remove the code then? Also, since the backend validation is back up again, do we plan to uncomment/fix frontend validation?

This may be needed for a more thorough discussion, as this problem/solution is rather complex. Nevertheless, I think the notifications now correctly save input data.

File: apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.ts [hunk 1]
```diff
@@ -109,6 +109,24 @@ export class AddEditNotificationDialogComponent implements OnInit {
...
+        this.inputFormGroup.get(`${lang}-html-text`).value
+      );
+      this.applicationMail.htmlMessage[lang].text = escaped.escapedHtml;
+
+      // Update application with content from FormControl
+      this.applicationMail.message[lang].subject = this.inputFormGroup.get(
+        `${lang}-plain-subject`
+      ).value;
+      this.applicationMail.message[lang].text = this.inputFormGroup.get(`${lang}-plain-text`).value;
+    }
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.ts:128]: Please extract this logic so it's not duplicated for both save and create.

File: apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.html [hunk 2]
```diff
@@ -149,11 +149,15 @@ <h1 mat-dialog-title>{{'DIALOGS.NOTIFICATIONS_ADD_EDIT_MAIL.TITLE_EDIT' | transl
                             *ngIf="format === 'plain'"
                             formControlName="{{lang}}-plain-subject"
                             (focus)="isTextFocused = false"
+                            [value]="this.applicationMail.message[lang].subject"
+                            (input)="this.applicationMail.message[lang].subject = inputFormGroup.controls[lang + '-plain-subject'].value"
```
Comment (bodnara) [inline apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.html:153]: I believe this should not be needed anymore. Also the way I understand it, is that the form control serves exactly this purpose of input-value connection (with some additional features).

Comment (mattjoke) [reply apps/admin-gui/src/app/shared/components/dialogs/add-edit-notification-dialog/add-edit-notification-dialog.component.html]: Yes. Once I removed the duplicities, the problem no longer occurs.

**Review by bodnara — APPROVED** [review]

**Review by xflord — APPROVED** [review]
gj

**Pull request merged**